zkevm/src/circuit/builder.rs

```diff
@@ -49,29 +49,37 @@ fn extend_address_to_h256(src: &Address) -> [u8; 32] {
     bts.as_slice().try_into().expect("32 bytes")
 }
 
-pub fn block_result_to_witness_block<F: Field>(
+pub fn block_result_to_witness_block(
     block_result: &BlockResult,
 ) -> Result<Block<Fr>, anyhow::Error> {
-    let chain_id = if let Some(tx_trace) = block_result.block_trace.transactions.get(0) {
+    block_results_to_witness_block(&[block_result.clone()])
```

Comment:
Could also use `slice::from_ref` here.

```suggestion
    block_results_to_witness_block(std::slice::from_ref(block_result))
```